gax/test/unit/streaming.ts

```diff
@@ -1466,6 +1474,67 @@ describe('streaming', () => {
       done();
     });
   });
+
+  it('properly emits the end event at the end of a pipeline transformation', done => {
+    const spy = sinon.spy((...args: Array<{}>) => {
+      assert.strictEqual(args.length, 3);
+      const s = new PassThrough({
+        objectMode: true,
+      });
+      s.push({resources: [1, 2]});
+      s.push(null);
+      setImmediate(() => {
+        s.emit('metadata');
+      });
+      return s;
```

Comment:
I think you should probably also be emitting "status" after metadata to properly be imitating a grpc stream